src/widgets/RealtimeEffectPanel.cpp

```diff
@@ -0,0 +1,711 @@
+         {
+            if(auto boxSizer = dynamic_cast<wxBoxSizer*>(parent->GetSizer()))
+            {
+               if(boxSizer->GetOrientation() == wxVERTICAL)
+               {
+                  auto srcItem = boxSizer->GetItem(mSourceIndex);
+
+                  const auto proportion = srcItem->GetProportion();
+                  const auto flag = srcItem->GetFlag();
+                  const auto border = srcItem->GetBorder();
+                  
+                  boxSizer->Remove(mSourceIndex);
+                  boxSizer->Insert(
+                     mTargetIndex < mSourceIndex ? mTargetIndex : mTargetIndex - 1,
+                     this,
+                     proportion,
+                     flag,
+                     border
+                  );
+               }
+            }
+         }
+         DragFinished();
+      }
+      
+      void OnMove(wxMouseEvent& evt)
+      {
+         if(!mDragging)
+            return;
+
+         auto parent = GetParent();
+         if(!parent)
+            return;
+
+         wxPoint newPosition = wxGetMousePosition() - mInitialPosition;
+         Move(GetParent()->ScreenToClient(newPosition));
+
+         if(auto boxSizer = dynamic_cast<wxBoxSizer*>(parent->GetSizer()))
+         {
+            if(boxSizer->GetOrientation() == wxVERTICAL)
+            {
+               auto targetIndex = mSourceIndex;
+               
+               //assuming that items are ordered from top to bottom (i > j <=> y(i) > y(j))
+               //compare wxSizerItem position with the current MovableControl position!
+               if(GetPosition().y < boxSizer->GetItem(mSourceIndex)->GetPosition().y)
+               {
+                  //moving up
+                  for(int i = 0; i < mSourceIndex; ++i)
+                  {
+                     const auto item = boxSizer->GetItem(i);
+                     
+                     if(GetRect().GetTop() <= item->GetPosition().y + item->GetSize().y / 2)
+                     {
+                        targetIndex = i;
+                        break;
+                     }
+                  }
+               }
+               else
+               {
+                  //moving down
+                  for(int i = static_cast<int>(boxSizer->GetItemCount()) - 1; i > mSourceIndex; --i)
+                  {
+                     const auto item = boxSizer->GetItem(i);
+                     if(GetRect().GetBottom() >= item->GetPosition().y + item->GetSize().y / 2)
+                     {
+                        targetIndex = i + 1;
+                        break;
+                     }
+                  }
+               }
+
+               if(targetIndex != mTargetIndex)
+               {
+                  mTargetIndex = targetIndex;
+                  PostDragEvent(parent, EVT_MOVABLE_CONTROL_DRAG_POSITION);
+               }
+            }
+         }
+      }
+   };
+
+   wxDEFINE_EVENT(EVT_EFFECT_ITEM_REMOVE, wxCommandEvent);
```

Comment:
An alternative possibility:  Don't define this event.  Don't post it from RealtimeEffectControl.

Instead, make the RealtimeEffectList s `Publisher` of messages when its item list change -- analogous to what TrackList does.  (Don't delay events with CallAfter as TrackList does for legacy reasons.  Let listeners CallAfter to delay their reactions as they need.)

Then the RealtimeEffectsWindow subscribes.

This way of doing it is more consistent with existing designs and would allow other subscribers if we ever need them, which won't need to cooperate specially with RealtimeEffectControl.

Reply:
I like this solution better for the possible future expansions.  But you can do just enough for preset needs:  let the list publish only deletions.  We might generalize for additions and permutations later.

Reply:
It is not much sense in doing so: `MovableControl` is a wxWidgets based component

Reply:
I do not understand your response.  I think you did not understand my objection.

Changes in the RealtimeEffectList — a UI neutral internal entity— should be published with our UI neutral Publisher class, as is also done for changes of the TrackList.  Then this new UI works more indirectly:  one drag and drop changes the RealtimrEffectList object, then it publishes its message, then the other control reacts.  But then it is also possible for other things that need to be kept consistent with the real-time effect list to react also.

I do not want to use Qt signals and slots outside of a UI layer of code built on a toolkit neutral layer.